langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Split.java

```diff
@@ -33,12 +33,14 @@
 public class Split {
 
     public static BArray split(BRegexpValue regExp, BString str) {
+        String[] splitStrArr;
         String originalString = str.getValue();
         RegExpValue translatedRegExpVal = RegExpFactory.translateRegExpConstructs((RegExpValue) regExp);
         String regex = StringUtils.getStringValue(translatedRegExpVal, null);
-        String[] splitStrArr = originalString.split(regex);
-        if (splitStrArr.length == 0) {
-            splitStrArr = new String[]{originalString};
+        if (regex.isEmpty()) {
```

Comment:
The behaviour of split function for empty regex is different in Java and Java script.

--- Java ---

```
String[] a = "abab".split("", -1); // ["a", "b", "a", "b", ""]
String[] a = "abab".split(""); // ["a", "b", "a", "b"]
```

--- Javascript

```
const re1 = new RegExp("");
var a = "abab".split(re1);
console.log("Length:- ", a.length);  // Length:-  4
console.log(a); // [ 'a', 'b', 'a', 'b' ]
```

This is the reason to add this if condition in here.
But this if condition only handles the `""` kind of empty regex.
There are few other empty regex usages that does not handle by this. ex:- `a|`, `()`, So currently figuring out a way to handle those.

Reply:
Created an issue. https://github.com/ballerina-platform/ballerina-lang/issues/39826